src/components/auth/registration/SignUpForm.jsx

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import * as registrationActions from '../../../actions/auth/registrationAction';
+import '../../../assets/scss/signup.scss';
+import Button from './Button';
+import Input from './Input';
+import Loading from './Loading';
+
+export class SignUpForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      email: '',
+      username: '',
+      password: '',
+      emailError: '',
+      usernameError: '',
+      passwordError: '',
+      confirmpassword: '',
+      confirmPasswordError: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.submitForm = this.submitForm.bind(this);
+    this.validateFormData = this.validateFormData.bind(this);
+    this.handleClearForm = this.handleClearForm.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      [event.target.name]: event.target.value,
+    }, () => { this.validateFormData(); });
+  }
+
+  validateFormData() {
+    const {
+      email,
+      username,
+      password,
+      confirmpassword,
+    } = this.state;
+
+    // eslint-disable-next-line no-useless-escape
+    const passwordValidation = new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})');
+    // eslint-disable-next-line no-useless-escape
+    const usernameValidation = new RegExp('^([a-zA-Z\d]+[-_])*[a-zA-Z\d*]+$');
+
+    this.setState({
+      emailError: email.length > 3 ? null : 'Email must be longer than 3 characters',
+      usernameError: username.length > 3 && usernameValidation.test(username) ? null : 'Username should be 4 characters and cannot be integers, have white spaces or symbol',
+      passwordError: passwordValidation.test(password) ? null : 'Password should be 8 to more characters with atleast a number, capital and small letter.',
+      confirmPasswordError: confirmpassword === password ? null : 'Passwords don\'t match',
+    });
+  }
+
+
+  submitForm(event) {
+    event.preventDefault();
+    const { email, username, password } = this.state;
+    const { actions } = this.props;
+    const newUser = { email, username, password };
+    actions.registerUser(newUser);
+    this.handleClearForm(event);
+  }
+
+  handleClearForm(event) {
```

Comment:
Do you really need to pass the event here?

Reply:
This is resolved.